### PR TITLE
Fix epoll

### PR DESCRIPTION
**Pull request by rileyberton**

### CODE REVIEW

**Comment by aidancully** [comment]
LGTM

File: src/eventer/eventer_epoll_impl.c [hunk 1]
```diff
@@ -287,31 +287,51 @@ static void eventer_epoll_impl_trigger(eventer_t e, int mask) {
...
+       *
+       * Treat this special case the same as a cross thread trigger
+       * and just queue this event to be picked up on the next loop
+       */
       eventer_cross_thread_trigger(e, mask);
       return;
     }
-    release_master_fd(fd, lockstate);
+    /*
+     * If we are readding the event to the master list here, also do the needful
```
Comment (postwait) [inline src/eventer/eventer_epoll_impl.c]: needful?

File: src/eventer/eventer_epoll_impl.c [hunk 2]
```diff
@@ -287,31 +287,51 @@ static void eventer_epoll_impl_trigger(eventer_t e, int mask) {
...
+    struct epoll_event _ev;
+    memset(&_ev, 0, sizeof(_ev));
+    _ev.data.fd = fd;
+    spec = eventer_get_spec_for_event(e);
+    if(mask & EVENTER_READ) _ev.events |= (EPOLLIN|EPOLLPRI);
+    if(mask & EVENTER_WRITE) _ev.events |= (EPOLLOUT);
+    if(mask & EVENTER_EXCEPTION) _ev.events |= (EPOLLERR|EPOLLHUP);
+
+    mtevL(eventer_deb, "epoll_ctl(%d, add, %d)\n", spec->epoll_fd, fd);
+    if (epoll_ctl(spec->epoll_fd, EPOLL_CTL_ADD, fd, &_ev) != 0) {
```
Comment (postwait) [inline src/eventer/eventer_epoll_impl.c]: What if `mask & (EVENTER_{READ,WRITE,EXCEPTION}) == 0`... we'll attempt to add "nothing".

Comment (rileyberton) [inline src/eventer/eventer_epoll_impl.c]: can mask be zero here?  We are triggered from something.. maybe an assertion that mask != 0?

File: src/eventer/eventer_epoll_impl.c (hunk 1 above)
Comment (rileyberton) [inline src/eventer/eventer_epoll_impl.c]: https://en.wikipedia.org/wiki/Do_the_needful

**Pull request merged**